### ISSUE TITLE
Incorrect behaviour of `tf.nn.sigmoid` on complex inputs

### ISSUE THREAD

**Issue by jshepherd01** [tensorflow/tensorflow#61800]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

tf v1.12.1-96406-gfa4d29bfef8 2.14.0-dev20230706

### Custom code

Yes

### OS platform and distribution

WSL Ubuntu 20.04.5 LTS

### Current behavior?

`tf.nn.sigmoid` gives incorrect values for complex numbers with large, negative real parts (which should map to within a rounding error of 0). For example, when `x = -709-1j` (`complex128`) I expect `tf.nn.sigmoid(x) = 0`, but instead `tf.nn.sigmoid(x) = 1`.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
x = tf.constant([-709-1j], dtype=tf.complex128)
tf.nn.sigmoid(x)  # <tf.Tensor: shape=(1,), dtype=complex128, numpy=array([1.+0.j])>
```

**Comment by sushreebarsa** [comment]
@sachinprasadhs I was able to replicate the issue on colab using TF v2.13, tf-nightly. Please find the [gist](https://colab.research.google.com/gist/sushreebarsa/ea440d43307ea574e87511ad26ac5c12/61800.ipynb) here for reference. Thank you!

**Comment by dlibk** [comment]
The same issue happens on `tf.Tensor((-inf+0j), shape=(), dtype=complex128)`, the result is `1+0j` instead of 0.
Please see the code below.
```python
input = tf.constant(-np.inf, dtype='complex128')
print(input)  # tf.Tensor((-inf+0j), shape=(), dtype=complex128)

out = tf.nn.sigmoid(input)
print(out)  # tf.Tensor((1+0j), shape=(), dtype=complex128)
```

**Comment by LakshmiKalaKadali** [comment]
Hi @jshepherd01,

The `tf.nn.sigmoid` is working as intended for complex inputs. Checked with different examples  in TF 2.17 and TF 2.18.0. 
```
test_inputs = [
        tf.constant([-np.inf], dtype=tf.complex128),
        tf.constant([-709-1j], dtype=tf.complex128),
        tf.constant([-709+1j], dtype=tf.complex128),
        tf.constant([-1-1j], dtype=tf.complex128),
        tf.constant([-1+1j], dtype=tf.complex128),
        tf.constant([0-1j], dtype=tf.complex128),
        tf.constant([0+1j], dtype=tf.complex128),
        
    ]
    
print("Complex Sigmoid Investigations:")
for x in test_inputs:
  result = tf.nn.sigmoid(x)
  print(f"\nInput: {x.numpy()}")
  print(f"Sigmoid Output: {result.numpy()}")
```
OUTPUT
```Input: [-inf+0.j]
Sigmoid Output: [0.+0.j]

Input: [-709.-1.j]
Sigmoid Output: [0.-0.j]

Input: [-709.+1.j]
Sigmoid Output: [0.+0.j]

Input: [-1.-1.j]
Sigmoid Output: [0.21795843-0.20194823j]

Input: [-1.+1.j]
Sigmoid Output: [0.21795843+0.20194823j]

Input: [0.-1.j]
Sigmoid Output: [0.5-0.27315124j]

Input: [0.+1.j]
Sigmoid Output: [0.5+0.27315124j]
```
Please refer to the [gist ](https://colab.sandbox.google.com/gist/LakshmiKalaKadali/c334ec93551887004f64d47084667a23/tf_61800.ipynb) as well.

Thank You
